packages/subgraph/src/v1-analytics/mapping.ts

```diff
@@ -194,19 +220,26 @@ export function handleTransactionFulfilled(event: TransactionFulfilled): void {
 
   transaction!.save();
 
-  // router receives liquidity back on sender fulfill
+  // Get asset balance
+  let assetBalanceId = transaction.sendingAssetId.toHex() + "-" + event.params.router.toHex();
+  let assetBalance = AssetBalance.load(assetBalanceId);
+  if (assetBalance == null) {
+    assetBalance = new AssetBalance(assetBalanceId);
+    assetBalance.assetId = transaction.sendingAssetId;
+    assetBalance.router = event.params.router.toHex();
+    assetBalance.amount = new BigInt(0);
+    assetBalance.supplied = new BigInt(0);
+    assetBalance.locked = new BigInt(0);
+  }
+
   if (transaction.chainId == transaction.sendingChainId) {
-    let assetBalanceId = transaction.sendingAssetId.toHex() + "-" + event.params.router.toHex();
-    let assetBalance = AssetBalance.load(assetBalanceId);
-    if (assetBalance == null) {
-      assetBalance = new AssetBalance(assetBalanceId);
-      assetBalance.assetId = transaction.sendingAssetId;
-      assetBalance.router = event.params.router.toHex();
-      assetBalance.amount = new BigInt(0);
-    }
+    // router receives liquidity back on sender fulfill
     assetBalance.amount = assetBalance.amount.plus(transaction.amount);
-    assetBalance.save();
+  } else {
+    // router releases locked liquidity on receiver fulfill
+    assetBalance.locked = assetBalance.locked.sub(transaction.amount);
   }
+  assetBalance.save();
 
   // update metrics
   let timestamp = event.block.timestamp.toI32();
```

Comment:
Ik you didn't write this code, but see my comment about calculating the `dayStartTimestamp` and compare that to the `hourStartTimestamp` below

https://github.com/connext/nxtp/pull/570/files#r759804355

Reply:
https://github.com/connext/nxtp/pull/570#discussion_r759807650